requirements.txt

```diff
@@ -3,7 +3,7 @@ codetiming==1.3.0
 cryptography==36.0.2
 Django==2.2.27
 dj-database-url==0.5.0
-django-allauth==0.39.1
+django-allauth==0.50.0
```

Comment:
`ACCOUNT_RATE_LIMITS` is introduced which uses per minute (guessing from this [code snippet](https://github.com/pennersr/django-allauth/blob/8a4201ade850a0819513cdbff50f251087e4651b/allauth/ratelimit.py#L12-L22)). It seems well have to provide `429.html` as mentioned [here](https://django-allauth.readthedocs.io/en/latest/release-notes.html?highlight=rate%20limiting#backwards-incompatible-changes-1).

Reply:
I'm not sure how rate-limited interacts with the React front-end. Before I re-enabled `SOCIALACCOUNT_LOGIN_ON_GET=True`, login involved a barely-styled auth form. 

I think a `429.html` is the right solution, so it is a usable error rather than a "template not found error". I'll spend an hour playing around and see if I can get rate-limited locally.

Reply:
TLDR: I'm adding a `429.html` which I don't think will ever render.

I went through the steps to get this working. In one shell:

```sh
brew install redis
/opt/homebrew/opt/redis/bin/redis-server /opt/homebrew/etc/redis.conf
```

In my `.env`, I added:
```sh
REDIS_URL="redis://127.0.0.1:6379"
```

I tested it:
```sh
pip install -r requirements.txt
./manage.py shell
>>> from django.core.cache import cache
>>> cache.set('foo', 'bar')
True
>>> cache.get('foo')
'bar'
```

And then added to settings:

```python
ACCOUNT_RATE_LIMITS = {
    "change_password": "1/d",
    "manage_email": "1/d",
    "reset_password": "1/d",
    "reset_password_email": "1/d",
    "reset_password_from_key": "1/d",
    "signup": "1/d",
}
ACCOUNT_LOGIN_ATTEMPTS_LIMIT=1
ACCOUNT_LOGIN_ATTEMPTS_TIMEOUT=300
```

I then tried to signup multiple times, and no rate limiting. 

I peeked at the keys:

```sh
> redis-cli keys \*
1) ":1:foo"
2) ":1:django.contrib.sessions.cachej8jyv03r2r7mcxvgcdnrxjabfrs7gkzt"
```

So, despite a cache being setup and a rate of once a day, allauth is not storing any usage data.

I peeked at the code:

https://github.com/pennersr/django-allauth/blob/d367458580f5ebe6e6a0903ba78f85e2e082e0f2/allauth/ratelimit.py#L58-L60

```python
if request.method == "GET" or not amount or not duration:
        pass
```

SO, rate limiting is only in effect if the request is a `POST`, and since we use `GET` for sign-in and sign-ups, django-allauth doesn't keep track of it.

I wrote a basic ``429.html`` that doesn't use Fluent or our website styling, so it will be very basic. I also think it will not ever be displayed.